core/src/main/java/org/apache/gravitino/listener/ModelEventDispatcher.java

```diff
@@ -0,0 +1,400 @@
+    this.eventBus = eventBus;
+    this.dispatcher = dispatcher;
+  }
+
+  /**
+   * Register a model in the catalog if the model is not existed, otherwise the {@link
+   * ModelAlreadyExistsException} will be thrown. The {@link Model} object will be created when the
+   * model is registered, users can call {@link ModelCatalog#linkModelVersion(NameIdentifier,
+   * String, String[], String, Map)} to link the model version to the registered {@link Model}.
+   *
+   * @param ident The name identifier of the model.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param properties The properties of the model. The properties are optional and can be null or
+   *     empty.
+   * @return The registered model object.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws ModelAlreadyExistsException If the model already registered.
+   */
+  @Override
+  public Model registerModel(NameIdentifier ident, String comment, Map<String, String> properties)
+      throws NoSuchSchemaException, ModelAlreadyExistsException {
+    // TODO: preEvent
+    try {
+      Model model = dispatcher.registerModel(ident, comment, properties);
+      ModelInfo modelInfo = new ModelInfo(model);
+      eventBus.dispatchEvent(
+          new RegisterModelEvent(PrincipalUtils.getCurrentUserName(), ident, modelInfo));
+      return model;
+    } catch (Exception e) {
+      // TODO: failureEvent
+      throw e;
+    }
+  }
+
+  /**
+   * Register a model in the catalog if the model is not existed, otherwise the {@link
+   * ModelAlreadyExistsException} will be thrown. The {@link Model} object will be created when the
+   * model is registered, in the meantime, the model version (version 0) will also be created and
+   * linked to the registered model.
+   *
+   * @param ident The name identifier of the model.
+   * @param uri The model artifact URI.
+   * @param aliases The aliases of the model version. The aliases should be unique in this model,
+   *     otherwise the {@link ModelVersionAliasesAlreadyExistException} will be thrown. The aliases
+   *     are optional and can be empty. Also, be aware that the alias cannot be a number or a number
+   *     string.
+   * @param comment The comment of the model. The comment is optional and can be null.
+   * @param properties The properties of the model. The properties are optional and can be null or
+   *     empty.
+   * @return The registered model object.
+   * @throws NoSuchSchemaException If the schema does not exist when register a model.
+   * @throws ModelAlreadyExistsException If the model already registered.
+   * @throws ModelVersionAliasesAlreadyExistException If the aliases already exist in the model.
+   */
+  @Override
+  public Model registerModel(
+      NameIdentifier ident,
+      String uri,
+      String[] aliases,
+      String comment,
+      Map<String, String> properties)
```

Comment:
Why do we have these parameters?
I was thinking that we have a model registered in the first place and then hook some event dispatching logic to that model.
If a model is already there, why are we providing these parameters?